### DISCUSSION TITLE
How do I use unqualified names?

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/CsWin32#1012] [Q&A] [answered]
I want to use the CreateWaitableTimer function, but I'm getting a warning that the method name is ambiguous, and I should use the fully qualified name. But I just want to use the function as documented here [Using Waitable Timer Objects](https://learn.microsoft.com/en-us/windows/win32/sync/using-waitable-timer-objects).

**Comment by AArnott** [comment]
I don't know why CsWin32 is emitting that error, since it's supposed to default to the W suffixed variants. I'll look into it.
In the meantime though, at least in my testing it seems you only have to add the `W` suffix in NativeMethods.txt. In your .cs files, the suffix can be omitted.

**Reply by AArnott** [reply, marked as answer]
Ah, the reason is because the `CreateWaitableTimer` variants and related functions are in distinct namespaces:

![image](https://github.com/microsoft/CsWin32/assets/3548/38781503-be5f-4cfc-8b2f-7a54adcfd517)

I filed [a bug](https://github.com/microsoft/win32metadata/issues/1676) against the win32metadata repo to fix this.
